internal/dinosaur/pkg/workers/dinosaurmgrs/dinosaurs_auth_config.go

```diff
@@ -0,0 +1,102 @@
+package dinosaurmgrs
+
+import (
+	"github.com/golang/glog"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/stackrox/acs-fleet-manager/internal/dinosaur/pkg/api/dbapi"
+	"github.com/stackrox/acs-fleet-manager/internal/dinosaur/pkg/config"
+	"github.com/stackrox/acs-fleet-manager/internal/dinosaur/pkg/services"
+	"github.com/stackrox/acs-fleet-manager/pkg/workers"
+)
+
+// CentralAuthConfigManager updates CentralRequests with auth configuration.
+type CentralAuthConfigManager struct {
+	workers.BaseWorker
+	centralService services.DinosaurService
+	centralConfig  *config.CentralConfig
+}
+
+var _ workers.Worker = &CentralAuthConfigManager{}
+
+func NewCentralAuthConfigManager(centralService services.DinosaurService, centralConfig *config.CentralConfig) *CentralAuthConfigManager {
+	return &CentralAuthConfigManager{
+		BaseWorker: workers.BaseWorker{
+			ID:         uuid.New().String(),
+			WorkerType: "central_auth_config",
+			Reconciler: workers.Reconciler{},
+		},
+		centralService: centralService,
+		centralConfig:  centralConfig,
+	}
+}
+
+// Start uses base's Start()
+func (k *CentralAuthConfigManager) Start() {
+	k.StartWorker(k)
+}
+
+// Stop uses base's Stop()
+func (k *CentralAuthConfigManager) Stop() {
+	k.StopWorker(k)
+}
+
+// Reconcile fetches all CentralRequests without auth config from the DB and
+// updates them.
+func (k *CentralAuthConfigManager) Reconcile() []error {
+	glog.Infoln("reconciling auth config for Centrals")
+	var errs []error
+
+	centralRequests, listErr := k.centralService.ListDinosaursWithoutAuthConfig()
+	if listErr != nil {
+		errs = append(errs, errors.Wrap(listErr, "failed to list centrals without auth config"))
+	} else {
+		glog.V(5).Infof("%d central(s) need auth config to be added", len(centralRequests))
+	}
+
+	for _, cr := range centralRequests {
+		if err := augmentWithAuthConfig(cr, k.centralConfig); err != nil {
```

Comment:
```suggestion
		if err := k.augmentWithAuthConfig(cr); err != nil {
```
seems more fitting to turn that into a method?

Reply:
Nvm, I see now why it isn't one.